zvedamrance/src/main/java/UserDao.java

```diff
@@ -0,0 +1,124 @@
+import java.util.Scanner;
+
+public class UserDao {
+    private static final String CREATE_USER_QUERY =
+            "INSERT INTO users(username, email, password) VALUES (?, ?, ?)";
+    private static final String SELECT_USER_BY_ID =
+            "SELECT * FROM users WHERE id = ?";
+    private static final String UPDATE_USER =
+            "UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?";
+    private static final String DELETE_USER =
+            "DELETE FROM users WHERE id = ?";
+    private static final String SELECT_ALL =
+            "SELECT * FROM users";
+
+
+    public String hashPassword(String password) {
+        return BCrypt.hashpw(password, BCrypt.gensalt());
+    }
+
+    public User create(User user) {
+        try (Connection conn = DbUtil.getConnection()) {
+            final PreparedStatement statement = conn.prepareStatement(CREATE_USER_QUERY, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.executeUpdate();
+            final ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getInt(1));
+            }
+            System.out.println("Created user: " + user.toString());
+            return user;
+        } catch (SQLException e) {
+            System.out.println("Error, user creation failed.");
+            e.printStackTrace();
+            return null;
+        }
+    }
+    public User read(int userId) {
+        try (Connection conn = DbUtil.getConnection()) {
+            final PreparedStatement statement = conn.prepareStatement(SELECT_USER_BY_ID);
+            statement.setInt(1, userId);
+            final ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String username = resultSet.getString("username");
+                String email = resultSet.getString("email");
+                String password = resultSet.getString("password");
+                System.out.println("Found user with user ID " + userId);
+                return new User(userId,username, email, password);
+            } else {
+                System.out.println("User with id " + userId + " not found.");
+                return null;
+            }
+        } catch (SQLException e) {
+            System.out.println("Error, user read failed.");
+            throw new RuntimeException(e);
+        }
+    }
+    public void update(User user) {
+        try (Connection conn = DbUtil.getConnection(); Scanner sc = new Scanner(System.in)) {
+
+            final PreparedStatement statement = conn.prepareStatement(UPDATE_USER);
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
```

Comment:
BUG: zrus hashovani hesla pri updatu, protoze stare heslo by si znova zaheshoval.

```suggestion
statement.setString(3, user.getPassword());
```

Reply:
opraveno